### PR TITLE
Bitrunning 1.5: Secondary Objectives

### PR DESCRIPTION
**Pull request by Thunder12345**
## About The Pull Request

Added secondary objective lockboxes to bitrunning. These pull from a list of secondary objective loot on the domain, with a limited quantity of items. Once there are no items left to pull, the secondary objective disappears. If multiple secondary objective markers are placed, they will be placed until all markers have been used, or all the items in the loot pool are already spoken for.

To support this functionality, adds SSbitrunning, which stores all domains as instances, instead of checking the hardcoded types as previously. SSbitrunning manages listing domains for the quantum console, and rolling secondary loot.

As an example of this functionality, added a side path to Glacier Grind with a polar bear and some loot.

## Why It's Good For The Game

Secondary objectives give mappers ways to encourage players to venture into a wider range of domains by offering non-trivial loot beyond the fluff items given in the main caches. The absolute limit on the number of items available ensures these items can't be farmed.

As well as supporting secondary objectives, SSbitrunning allows for future support of features relying on mid-round modification of domains, for instance adding custom domains.

## Changelog
:cl:
add: Added secondary objectives to bitrunning!
add: Pick up encrypted curiosities and return them to the safehouse to claim their contents.
add: Glacier Grind has been given a secondary objective, look out for the limited edition hat.
add: Bitrunning domains can now be modified during the round by admins.
/:cl:

### CODE REVIEW

**Comment by Admiral-Gigaton** [comment]
ssbitrunning, more like sshitrunning!

haha, just kidding, i think this is a great idea

File: code/modules/bitrunning/objects/loot_box.dm [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+		log_runtime("Decrypted curiosity was created with an invalid source domain. [completed_domain.name] ([completed_domain.type]).")
+		return INITIALIZE_HINT_QDEL
+
+	source_domain = completed_domain
+
+	. = ..()
+	atom_storage.max_specific_storage = WEIGHT_CLASS_NORMAL
+	atom_storage.max_slots = 1
+	atom_storage.max_total_storage = 3
+	atom_storage.locked = STORAGE_NOT_LOCKED
```
Comment (JohnFulpWillard) [inline code/modules/bitrunning/objects/loot_box.dm:47]: Since crates are lockable, if you lock this can it be re-unlocked or nah?

Comment (Thunder12345) [reply code/modules/bitrunning/objects/loot_box.dm:47]: That’s a thing? Good question

File: code/modules/bitrunning/objects/byteforge.dm [hunk 2]
```diff
@@ -63,5 +63,5 @@
 /obj/machinery/byteforge/proc/start_to_spawn(obj/cache)
 	flicker()
 
-	addtimer(CALLBACK(src, PROC_REF(spawn_cache), cache), 1 SECONDS, TIMER_UNIQUE|TIMER_OVERRIDE|TIMER_STOPPABLE)
+	addtimer(CALLBACK(src, PROC_REF(spawn_cache), cache), 1 SECONDS, TIMER_STOPPABLE)
```
Comment (Ghommie) [inline code/modules/bitrunning/objects/byteforge.dm:66]: Without a reference to its id, a stoppable timer cannot be stopped anyway. Is the flag even necessary now?

File: code/modules/mob/living/basic/space_fauna/bear/_bear.dm [hunk 3]
```diff
@@ -82,6 +82,11 @@
 	icon_dead = "snowbear_dead"
 	desc = "It's a polar bear, in space, but not actually in space."
 
+/mob/living/basic/bear/snow/ancient
+	name = "ancient polar bear"
+	desc = "A grizzled old polar bear, its hide thick enough to make it impervious to almost all weapons."
+	status_flags = CANPUSH | GODMODE
```
Comment (Ghommie) [inline code/modules/mob/living/basic/space_fauna/bear/_bear.dm:88]: just to be sure, explicitly state that these are not spawnable by gold cores.

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Pull request merged**